tasks/gabrielsebrao/PJA-17/listOfInternetProfile.kt

```diff
@@ -0,0 +1,36 @@
+class Person(val name: String, val age: Int, val hobby: String?, val referrer: Person?) {
+    fun showProfile() {
+        println("Name: ${this.name}")
+        println("Age: ${this.age}")
+        
+        if(hobby != null) {
+            print("Likes to ${this.hobby}.")
+        } else {
+            print("Doesn't have a hobby.")
+        } 
+        
+        if(referrer == null) {
+            print(" Doesn't have a referrer.")
+        } else if(referrer.hobby == null){
+            print(" Has a referrer named ${referrer.name}, who doesn't have a hobbie.")
+        } else {
+            print(" Has a referrer named ${referrer.name}, who likes to ${referrer.hobby}.")
+        }
+        
+        println("\n")
+    }
+}
+
+fun main() {
+    val people = listOf(
+        Person("Gabriel", 17, "play games", null),
+        Person("Daniel", 21, null, null),
+        Person("Nicolas", 30, "draw", null),
+        Person("Giulia", 18, "do the nails", null),
+        Person("Maira", 16, "design", null)
+    )
```

Comment:
in this case, we dont have a variable for each person, so we can't insert a referrer by just accessing it's refference. Adjust it so we can instantiate the list this way:

`val people = listOf(Gabriel,  Daniel, Nicolas, Giulia, Maira)`

Reply:
okay, now it's correct, thank you.

     val gabriel = Person("Gabriel", 17, "play games", null)
     val daniel = Person("Daniel", 21, null, gabriel)
     val nicolas = Person("Nicolas", 30, "draw", daniel)
     val giulia = Person("Giulia", 18, "do the nails", nicolas)
     val maira = Person("Maira", 16, "design", null)

     val people = listOf(gabriel, daniel, nicolas, giulia, maira)